task/fbc-related-image-check/0.1/fbc-related-image-check.yaml

```diff
@@ -0,0 +1,68 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "appstudio, hacbs"
+  name: fbc-related-image-check
+spec:
+  params:
+    - name: IMAGE_URL
+      description: the fully qualified image name
+  results:
+    - name: HACBS_TEST_OUTPUT
+  workspaces:
+    - name: workspace
+  steps:
+    - name: check-related-images
+      image: quay.io/redhat-appstudio/hacbs-test:latest
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      securityContext:
+        runAsUser: 1000
+      env:
+        - name: HOME
+          value: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      resources:
+        limits:
+          memory: 4Gi
+          cpu: 2
+        requests:
+          memory: 512Mi
+          cpu: 10m
+      script: |
+        #!/usr/bin/env bash
+        #Wanted to use extracted content from previous task, but for some ungodly reason you cannot create folder in workdir root directory
+        #unless its an emptydir volume wich is not transitioned to next step, because it is - emptydir
+        #I love tekton <3
+        mkdir confdir
+
+        conffolder=$(cat ../sanity-inspect-image/image_inspect.json | jq -r '.Labels ."operators.operatorframework.io.index.configs.v1"')
+        if ! oc image extract $(params.IMAGE_URL) --path $conffolder/*:confdir/ ; then
```

Comment:
Using `$(params.*)` in script blocks is almost always a bad idea. If the user can set the parameter value, they can always inject arbitrary code into the script.

https://tekton.dev/docs/pipelines/tasks/#substituting-in-script-blocks

As suggested in the Tekton docs, should we pass the params as env variables instead? Like the git-clone task does it: https://github.com/redhat-appstudio/build-definitions/blob/258144d6b15f193eb8a007551eef018d7c62afd8/task/git-clone/0.1/git-clone.yaml#L85-L91

Reply:
Agreed, updated according to recommended way.